lib/functional-constraints/searchOrCreateKeys.js

```diff
@@ -11,14 +11,15 @@ const validateSearchOrCreateKeys = (definition) => {
   const errors = [];
   const searchKeys = _.keys(definition.searches);
```

Comment:
I see now. Perhaps change `definition.searches[searchKey]` to: `~searchKeys.indexOf(searchKey)`